ci_framework/roles/os_must_gather/tasks/main.yml

```diff
@@ -61,8 +61,23 @@
           oc adm must-gather --image {{ cifmw_os_must_gather_image }}
           --timeout {{ cifmw_os_must_gather_timeout }}
           --host-network={{ cifmw_os_must_gather_host_network }}
-          --dest-dir {{ cifmw_os_must_gather_output_dir }}/logs > {{ cifmw_os_must_gather_output_dir }}/logs/os_must_gather.log
+          --dest-dir {{ cifmw_os_must_gather_output_dir }}/logs
+          -- ADDITIONAL_NAMESPACES={{ cifmw_os_must_gather_additional_namespaces }} gather &> {{ cifmw_os_must_gather_output_dir }}/logs/os_must_gather.log
+
   rescue:
+    - name: Create oc_inspect log directory
+      ansible.builtin.file:
+        path: "{{ cifmw_os_must_gather_output_dir }}/logs/oc_inspect"
+        state: directory
+        mode: "0755"
+
+    - name: Append additional namespaces to the default list
+      ansible.builtin.set_fact:
+        cifmw_os_must_gather_namespaces: "{{ cifmw_os_must_gather_namespaces | default([]) + [ item ] }}"
```

Comment:
Just one comment, based on awful experiences we have. Keep in mind that if the user passes `cifmw_os_must_gather_namespaces` as a variable the set_fact is not going to work and the value that will be used in the following task will be the one set in the variable and not in the fact. That's because the precedence of variables and facts...

Reply:
If you do something like this you may be able to get rid off what I'm saying at the same time we do this in a single task:
```yaml
- name: Inspect the cluster after must-gather failure
  ignore_errors: true # noqa: ignore-errors
  environment:
    KUBECONFIG: "{{ cifmw_openshift_kubeconfig }}"
    PATH: "{{ cifmw_path }}"
  ci_script:
    output_dir: "{{ cifmw_os_must_gather_output_dir }}/artifacts"
    script: |
        oc adm inspect namespace/{{ item }} --dest-dir={{ cifmw_os_must_gather_output_dir }}/logs/oc_inspect
  loop: >-
    {{
      (
        cifmw_os_must_gather_namespaces | default([]) +
        (
          cifmw_os_must_gather_additional_namespaces |
          map('split', ',') |
          list
        )
      ) | unique
    }}
  

```

Reply:
ack, it makes sense and we do not want to expose such variable to external users because we know we need it to get the namespaces as a result of combining default + additional.
Thanks for the hint here, I'll check the version you're proposing here and propose something that fit what you're describing (even the same if it works, but I feel like it's cleaner see if can compute that variable outside the `loop` keywork:  I like this solution though!)